services/src/main/java/org/exoplatform/news/listener/NewsPublicationListener.java

```diff
@@ -32,19 +38,32 @@
 
 public class NewsPublicationListener extends Listener<CmsService, Node> {
 
-  private NewsService newsService;
+  private NewsService            newsService;
+
+  private SessionProviderService sessionProviderService;
+
 
   public NewsPublicationListener() {
     newsService = WCMCoreUtils.getService(NewsService.class);
+    sessionProviderService = WCMCoreUtils.getService(SessionProviderService.class);
   }
 
   public void onEvent(Event<CmsService, Node> event) throws Exception {
+    ExoContainer container = PortalContainer.getInstance();
+    RepositoryService repositoryService = container.getComponentInstanceOfType(RepositoryService.class);
+    SessionProvider systemProvider = SessionProvider.createSystemProvider();
+    sessionProviderService.setSessionProvider(null, systemProvider);
+    Session session = systemProvider.getSession(
+                                                repositoryService.getCurrentRepository()
+                                                                 .getConfiguration()
+                                                                 .getDefaultWorkspaceName(),
+                                                repositoryService.getCurrentRepository());
     if (AuthoringPublicationConstant.POST_CHANGE_STATE_EVENT.equals(event.getEventName())) {
       Node targetNode = event.getData();
       if (targetNode.isNodeType("exo:news") && targetNode.getProperty(StageAndVersionPublicationConstant.CURRENT_STATE).getString().equals(PublicationDefaultStates.PUBLISHED)) {
         News news = newsService.convertNodeToNews(targetNode, false);
         if (StringUtils.isEmpty(news.getActivities())) {
-          newsService.createNews(news);
+          newsService.createNews(news, session);
         }
       }
     }
```

Comment:
We need to close the session provider here

Reply:
done